lib/screen/bloc/accounts/accounts_bloc.dart

```diff
@@ -53,6 +53,39 @@ class AccountsBloc extends AuBloc<AccountsEvent, AccountsState> {
       emit(AccountsState(accounts: accounts));
     });
 
+    on<ChangeAccountOrderEvent>((event, emit) {
+      int newOrder = event.newOrder;
+      final oldOrder = event.oldOrder;
+      if (oldOrder == newOrder ||
+          state.accounts == null ||
+          oldOrder >= state.accounts!.length ||
+          newOrder >= state.accounts!.length) {
+        return;
+      }
+
+      if (oldOrder < newOrder) {
+        newOrder -= 1;
+      }
+      final List<Account> accounts = [...state.accounts!];
+      final Account account = accounts.removeAt(oldOrder);
+      accounts.insert(newOrder, account);
+      emit(state.copyWith(accounts: accounts));
+      add(SaveAccountOrderEvent(accounts: accounts));
+    });
+
+    on<SaveAccountOrderEvent>((event, emit) async {
+      final accounts = event.accounts;
+      for (int i = 0; i < event.accounts.length; i++) {
+        final account = accounts[i];
+        if (account.persona != null) {
+          await _cloudDB.addressDao.setAddressOrder(account.key, i);
+        } else {
+          await _cloudDB.connectionDao
+              .setConnectionOrder(account.accountNumber, i);
+        }
```

Comment:
consider using `transaction` for bulk update multiple rows at the same time to reduce read/write database.